src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    static final int NUMBER_COUNT = 10;
+    static final int FIGURE_COUNT = 5;
+
+    public int getRandomNumber() {
+        return new Random().nextInt(NUMBER_COUNT) + 1;
+    }
+
+    public Figure getRandomFigure() {
+        Figure isoscelesTrapezoid = new IsoscelesTrapezoid(new ColorSupplier().getRandomColor(),
+                getRandomNumber(), getRandomNumber(), getRandomNumber());
+        Figure circle = new Circle(new ColorSupplier().getRandomColor(), getRandomNumber());
+        Figure rightTriangle = new RightTriangle(new ColorSupplier().getRandomColor(),
+                getRandomNumber(), getRandomNumber());
+        Figure rectangle = new Rectangle(new ColorSupplier().getRandomColor(), getRandomNumber(),
+                getRandomNumber());
+        Figure square = new Square(new ColorSupplier().getRandomColor(), getRandomNumber());
+
+        Figure[] figures = {isoscelesTrapezoid, circle, rightTriangle, rectangle, square};
+        return figures[new Random().nextInt(FIGURE_COUNT)];
```

Comment:
You create 5 different figures and use only one of them. Let's only generate one figure based on a random number. (You can use switch-case here).